package-res/resources/web/pentaho/data/Filter.js

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./filter/IsEqual",
+  "./filter/IsIn",
+  "./filter/And",
+  "./filter/Or",
+  "./filter/Not"
+], function(IsEqual, IsIn, AndFilter, OrFilter, NotFilter) {
+  "use strict";
+
+
+  var RootFilter = OrFilter.extend({
+    constructor: function(filterSpec) {
+      if(filterSpec)
+        this.base(fromSpec(filterSpec));
+      else
+        this.base();
+    }
+  });
+
+  return {
+    // Leaf nodes
+    IsEqual: IsEqual,
+    IsIn: IsIn,
+    // Non-leaf nodes
+    Or: OrFilter,
+    And: AndFilter,
+    Not: NotFilter,
+    //Root: RootFilter,
+
+    /**
+     * Create a filter from a spec
+     * @param {Object} spec - Specification of a Filter
+     * @returns {*}
+     */
+    create: function (spec) {
+      return new RootFilter(spec);
+    }
+  };
+
+
+
+  function fromSpec(filterSpec) {
```

Comment:
Oh, I totally agree.
The .toSpec was not part of the print
The functionality came up at some stage due to the need of writing tests in predefined states.

I stopped the development of this implementation when the discussion on which spec format to use came up.
The question I would like to put is whether is it worthwhile to invest any more effort *now* assuming this mongoDB-like format.